### PR TITLE
Bump actions/upload-artifact to v4

### PR DESCRIPTION
**Pull request by stevepiercy**
See https://github.com/plone/volto/pull/6250#issuecomment-2286845833

### CODE REVIEW

**Comment by stevepiercy** [comment]
I restarted the same 5 failed jobs that failed in #6250.

Here's an example of the failed command.

```
/usr/bin/make start-test-acceptance-webserver-seamless
/opt/hostedtoolcache/node/18.20.4/x64/bin/npx wait-on --httpTimeout 20000 http-get://127.0.0.1:55001/plone http://127.0.0.1:3000/ http://localhost
/etc/bash.bashrc: line 7: PS1: unbound variable
/etc/bash.bashrc: line 7: PS1: unbound variable
/etc/bash.bashrc: line 7: PS1: unbound variable
/etc/bash.bashrc: line 7: PS1: unbound variable
/etc/bash.bashrc: line 7: PS1: unbound variable
/etc/bash.bashrc: line 7: PS1: unbound variable
/etc/bash.bashrc: line 1: docker-compose: command not found
make: *** [Makefile:267: start-test-acceptance-webserver-seamless] Error 127
/home/runner/work/_actions/cypress-io/github-action/v6/dist/index.js:[39](https://github.com/plone/volto/actions/runs/10375646984/job/28733005594?pr=6253#step:5:40)00
                error = new Error(`The process '${this.toolPath}' failed with exit code ${this.processExitCode}`);
                        ^

Error: The process '/usr/bin/make' failed with exit code 2
```

I have no clue how to figure out how to start the server correctly. Help!

**Comment by davisagli** [comment]
@stevepiercy `docker-compose` is deprecated and needs to be replaced with `docker compose`. I did this already on main and the 16.x.x branch, but missed 17.x.x

**Comment by stevepiercy** [comment]
@davisagli thanks, gotta run, will do tonight.

File: news/6253.internal [hunk 1]
```diff
@@ -0,0 +1 @@
+Bump actions/upload-artifact to v4. @stevepiercy
```
Comment (davisagli) [inline news/6253.internal:1]:
```suggestion
Bump actions/upload-artifact to v4 and replace `docker-compose` with `docker compose`. @stevepiercy, @davisagli
```

**Review by ichim-david — APPROVED** [review]
Thx guys it's getting hard to maintain 3 branches with fixes :)

**Pull request merged**